### PR TITLE
#179 Skipping evaluation for package-info file

### PR DESCRIPTION
**Pull request by amriteya**
These changes allow JavaMatch TU to skip evaluation for package-info file by default. It also allows users to set result to true.

### CODE REVIEW

File: butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/conditions/java/JavaMatch.java [hunk 1]
```diff
@@ -19,7 +19,9 @@
  * if they all are true. If the specified Java class file contains
  * more than one type, only the outer one will be considered
  * during evaluation. If it has none, the evaluation will result
- * in false and a warning be returned.
+ * in false and a warning be returned. If the Java class file is
+ * package-info file, it returns a warning with a default result as false.
```
Comment (fabiocarvalho777) [inline butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/conditions/java/JavaMatch.java]: Given the points you made in the issue, which are very valid, let's please rephrase this to:

`If the Java class file is named package-info.java, the file content will not be evaluated and false and a warning will be returned. The boolean result for package-info.java file is configurable though.`

File: butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/conditions/java/JavaMatch.java [hunk 2]
```diff
@@ -106,6 +132,15 @@ protected TUExecutionResult execution(File transformedAppFolder, TransformationC
             fileInputStream = new FileInputStream(javaClassFile);
             CompilationUnit compilationUnit = JavaParser.parse(fileInputStream);
 
+            compilationUnit.setStorage(javaClassFile.toPath());
```
Comment (fabiocarvalho777) [inline butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/conditions/java/JavaMatch.java]: Since we have already a hold of the file (`javaClassFile`), do we really need to manipulate `compilationUnit` to check if file is named `package-info.java`? What if we just do `javaClassFile.getName()`?

In the future, if JavaParser adds `isPackageInfo` then it would make sense to delegate that check to JavaParser, but not at this time.

Does that make sense?

File: butterfly-utilities/src/test/java/com/paypal/butterfly/utilities/conditions/java/JavaMatchTest.java [hunk 3]
```diff
@@ -69,6 +69,40 @@ public void noCompilationUnitTest() {
         Assert.assertEquals(executionResult.getWarnings().get(0).getMessage(), "This Java class file has no declared types: " + noCompilationUnitFile.getAbsolutePath());
     }
 
+    @Test
```
Comment (fabiocarvalho777) [inline butterfly-utilities/src/test/java/com/paypal/butterfly/utilities/conditions/java/JavaMatchTest.java]: Amrit, notice that in both of your tests the JavaMatch doesn't have any condition.

Please add a test where the JavaMatch TO has a `JavaCondition` that we would expect to return true, but, because the file is a Java package info, false and a warning are returned instead.

And a fourth test would be to the opposite, a JavaMatch with a JavaCondition that is supposed to return false, but, because the file to be evaluated is a Java package info, and because we called `setPackageInfo(true)`, true and a warning are supposed to be returned instead.

Does that make sense?

Also, if it helps, you can create a parameterized helper method containing most of the boiler plate code, and all these `@Test` methods calling it. That is optional though, your call.

**Comment by amriteya** [comment]
@fabiocarvalho777 Addressed the comments and updated the PR.
